tools/StaticAnalysis/BreakingChangeAnalyzer/BreakingChangeAnalyzer.cs

```diff
@@ -0,0 +1,968 @@
+                        {
+                            bool foundKey = false;
+
+                            foreach (var newKey in newOutput.Type.Properties.Keys)
+                            {
+                                if (oldKey.Equals(newKey) && 
+                                    oldOutput.Type.Properties[oldKey].Name.Equals(newOutput.Type.Properties[newKey].Name))
+                                {
+                                    foundKey = true;
+                                    break;
+                                }
+                            }
+
+                            if (!foundKey)
+                            {
+                                sameOutput = false;
+                                break;
+                            }
+                        }
+
+                        if (sameOutput)
+                        {
+                            found = true;
+                        }
+                    }
+                }
+
+                // If we were unable to find the output type, log an issue
+                if (!found)
+                {
+                    issueLogger.LogBreakingChangeIssue(
+                        cmdlet: oldCmdlet,
+                        severity: 0,
+                        problemId: ProblemIds.BreakingChangeProblemId.ChangedOutputType,
+                        description: string.Format(Properties.Resources.ChangedOutputTypeDescription, 
+                            oldCmdlet.Name, oldOutput.Type.Name),
+                        remediation: string.Format(Properties.Resources.ChangedOutputTypeRemediation, 
+                            oldCmdlet.Name, oldOutput.Type.Name));
+                }
+            }
+        }
+
+        /// <summary>
+        /// Check if the default parameter set has changed, and if so, make sure
+        /// that the parameters are the same
+        /// </summary>
+        /// <param name="oldCmdlet"></param>
+        /// <param name="newCmdlet"></param>
+        /// <param name="issueLogger"></param>
+        private void CheckDefaultParameterName(
+            CmdletBreakingChangeMetadata oldCmdlet,
+            CmdletBreakingChangeMetadata newCmdlet,
+            ReportLogger<BreakingChangeIssue> issueLogger)
+        {
+            // If the default parameter name hasn't changed, or if there wasn't a
+            // default parameter set defined before, return
+            if (oldCmdlet.DefaultParameterSetName.Equals(newCmdlet.DefaultParameterSetName) ||
+                oldCmdlet.DefaultParameterSetName.Equals("__AllParameterSets"))
+            {
+                return;
+            }
+
+            ParameterSetMetadata oldDefaultParameterSet = null;
+            ParameterSetMetadata newDefaultParameterSet = null;
+
+            // Get the ParameterSetMetadata object corresponding to the
+            // default parameter set for the old assembly
+            foreach (var oldParameterSet in oldCmdlet.ParameterSets)
```

Comment:
Use linq instead to get the default parameter set using FirstOrDefault<T>(filterFuncrion)